core/src/main/java/org/jdbi/v3/core/config/ConfigCustomizer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.config;
+
+import org.jdbi.v3.meta.Beta;
+
+/**
+ * Customizes a given {@link ConfigRegistry}.
+ *
+ * @since 3.38.0
+ */
+@FunctionalInterface
+@Beta
+public interface ConfigCustomizer {
```

Comment:
I think I wrote a comment somewhere. This is not a consumer (we don't consume the config, we only customize it). And the code becomes really gnarly because there is a `Consumer<ConfigRegistry>` everywhere, which makes the code hard to read.